shotover-proxy/src/transforms/cassandra/sink_cluster/topology.rs

```diff
@@ -183,6 +227,174 @@ async fn fetch_current_nodes(
     Ok(new_nodes)
 }
 
+enum ReplicationStrategy {
+    SimpleStrategy {
+        replication_factor: usize,
+    },
+    NetworkTopologyStrategy {
+        datacenter_replication_factor: HashMap<String, usize>,
+    },
+    Other,
+}
+
+mod system_keyspaces {
+    use super::*;
+    use serde_json::{Map, Value as JsonValue};
+    use std::str::FromStr;
+
+    pub async fn query(
+        connection: &CassandraConnection,
+        data_center: &str,
+    ) -> Result<HashMap<String, KeyspaceMetadata>> {
+        let (tx, rx) = oneshot::channel();
+
+        connection.send(Message::from_frame(
+            Frame::Cassandra(CassandraFrame{
+                version: Version::V4,
```

Comment:
The topology task knows what version the protocol is using via the `let version = connection_info.connection_factory.get_version()?;` line.
We should use that version here.